pkg/kubernetes/kubernetes.go

```diff
@@ -94,6 +90,39 @@ func GetKubeConfig(ctx context.Context) (*restclient.Config, error) {
 	return config, nil
 }
 
+// getKubeConfigPath returns the Kubeconfig path from the environment variable KUBECONFIG.
+// If the kubeconfig flag is supplied as an argument to the process, then it overrides the value
+// obtained from the KUBECONFIG environment variable and uses the supplied flag value.
+func getKubeConfigPath(ctx context.Context) string {
+	log := logger.GetLogger(ctx)
+	// Get the Kubeconfig path from the environment variable
+	kubecfgPath := os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+	log.Infof("Kubeconfig path obtained from environment variable %q: %q",
+		clientcmd.RecommendedConfigPathEnvVar, kubecfgPath)
+
+	// Check if the kubeconfig flag is set
+	kubecfgFlag := flag.Lookup("kubeconfig")
+	if kubecfgFlag != nil {
+		flagValue := kubecfgFlag.Value.String()
+		if flagValue != "" {
+			kubecfgPath = flagValue
+			log.Infof("Kubeconfig path obtained from kubeconfig flag: %q", kubecfgPath)
+		} else {
+			log.Info("Kubeconfig flag is set but empty, using environment variable value")
```

Comment:
nit: `using` -> `checking`, since we'd still be checking if the env variable is set or not.

Reply:
done.